### ISSUE TITLE
Access to route for non-existent plugin

### ISSUE THREAD

**Issue by dalenewman** [mattermost/mattermost-plugin-calls#54]
I'm getting tons of these errors on my 6.4.2 mattermost, how do I make it stop?

```json
{
    "timestamp": "2022-04-15 16:56:39.235 Z",
    "level": "error",
    "msg": "Access to route for non-existent plugin",
    "caller": "app/plugin_requests.go:37",
    "missing_plugin_id": "com.mattermost.calls",
    "url": "/plugins/com.mattermost.calls/channels",
    "error": "plugin not found: com.mattermost.calls"
}
```

**Comment by streamer45** [comment]
Hi @dalenewman , 

is the plugin enabled and running? Those errors would suggest it's not but clients are still making requests to it.

**Comment by dalenewman** [comment]
No it’s not enabled.  There’s nothing about it in the config.json or the
system console (ui).

Is there perhaps something I can put in the server’s config to get the
clients to stop making requests for it?  For example, I could install it
but explicitly disable it?

Thanks.

**Comment by streamer45** [comment]
> No it’s not enabled. There’s nothing about it in the config.json or the system console (ui). Is there perhaps something I can put in the server’s config to get the clients to stop making requests for it? For example, I could install it but explicitly disable it? Thanks.

I see. While the error may be annoying it's also totally harmless. I think we need to figure out why the requests are happening with the plugin not even installed. To make them go away now you could install and enable the plugin but explicitly disable allowing calls so essentially it will make it unusable for users:

![image](https://user-images.githubusercontent.com/1832946/166109326-b398a52c-6575-4580-bc25-263689119a47.png)

Not sure it's worth it as the error is really just a log, nothing serious is happening.

@cpoile I am wondering if it's the mobile app that could be causing this?

**Comment by dalenewman** [comment]
Thanks, I will  give it a try.  It's worth it to me because from time to
time I have users complaining about Mattermost issues and I'd like to be
able to sift through the error log more easily (without so much error
noise).  I'll let you know how it works out and thanks for your help 👍

**Comment by streamer45** [comment]
Sure, that's fair enough. We'll also be looking into fixing this on the client side.

**Comment by streamer45** [comment]
Created [MM-43904](https://mattermost.atlassian.net/browse/MM-43904) to track this.

**Comment by cpoile** [comment]
Hi @dalenewman , I'm looking into this now. I've found where the problem is, and we'll discuss the solutions and get back to you with some options. Thanks!

**Comment by dalenewman** [comment]
Thanks @cpoile.

FWIW, here's my actions so far...

I've added the configuration for the calls plugin to _config.json_ with it is explicitly **disabled**.  In addition, I've added the plugin files (the plugin itself and the signature) to the _/opt/mattermost/prepackaged_plugins_ folder.  For some reason it doesn't automatically install (and I can't see any logs stating the reason why).

I was able to manually upload the plugin to my development environment and see the UI.  However, when I tried this in my integration testing environment, it crashed the Mattermost service.  So, I have opted _not_ to try and upload it in the production environment.

I am monitoring the logs to see if I get more errors.  The error is only present in the production environment.  So, I've directed the users to upgrade their clients just in case they're using an old client that's some how "remembering" the calls plugin from an earlier implementation.

Looking forward to hearing the options, because obviously (based on the above) I'm just using a trial and error strategy.

**Comment by streamer45** [comment]
Hi @dalenewman , the plugin should definitely not cause a crash. If it does please feel free to send some server logs our way (e.g. stacktrace) so we can have a look.


> I've directed the users to upgrade their clients just in case they're using an old client that's some how "remembering" the calls plugin from an earlier implementation.

Actually that may make things worse as it was a recent addition to the mobile app that's causing this.

The good news though is that @cpoile has sent a fix already which should be in the upcoming Mobile app release (both in 1.51.2 and 1.52.0). Once that's out you can ask your users to update which should finally eliminate the error logs.

**Comment by dalenewman** [comment]
Thanks so much 👍

**Comment by cpoile** [comment]
Hi @dalenewman, the fixed mobile app is in the app store now -- `v1.51.2`. `v1.52.0` will also have the fix when it's released later this month. Those users who upgrade will not generate the server messages you mentioned above. Thanks for your patience while we fixed it!

**Comment by streamer45** [comment]
Closing this as fixed.
